common/models/user.js

```diff
@@ -669,6 +669,42 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function beforeEmailUpdate(ctx, next) {
+      var oldEmail, newEmail, oldId, newId;
+      if (ctx.isNewInstance) return next();
+
+      if (ctx.where) {
+        ctx.hookState.id = ctx.where.id;
```

Comment:
Also `ctx.where.id` may not be set, consider the case of 

```js
User.updateAll(
  { where: { name: { 'Miroslav Bajtos' } } },
  { email: 'mbajtos@cz.ibm.com' });
```

You should use `ctx.where` as-is whent it's provided. After you get back results from the database, you need to change `ctx.where` to match only the `ids` found, to prevent a race condition where somebody could add/delete another record matching the original `ctx.where` but not included in your list of users.